controllers/tykoasapidefinition_controller.go

```diff
@@ -411,6 +622,41 @@ func deleteOasApi(ctx context.Context, tykOASCrd *v1alpha1.TykOasApiDefinition)
 	return nil
 }
 
+func (r *TykOasApiDefinitionReconciler) updateBaseOasAPIVersionsStatus(
+	ctx context.Context,
+	log logr.Logger,
+	tykOASCrd *v1alpha1.TykOasApiDefinition,
+) error {
+	if tykOASCrd == nil {
+		return nil
+	}
+
+	if tykOASCrd.Spec.Versioning.Enabled {
+		for _, v := range tykOASCrd.Spec.Versioning.Versions {
+			var versionOasAPI v1alpha1.TykOasApiDefinition
+
+			if err := r.Get(ctx, types.NamespacedName{
+				Name:      v.OasApiDefinitionRef,
+				Namespace: v.Namespace,
+			}, &versionOasAPI); err != nil {
+				if k8serrors.IsNotFound(err) {
+					log.Info("failed to get versioned TykOasApiDefinition CR", "err", err.Error())
+				}
+
+				return err
+			}
+
+			versionOasAPI.RemoveOASVersionStatus()
+
+			if err := r.Status().Update(ctx, &versionOasAPI); err != nil {
```

Comment:
Again, updating the status here might lead to conflicts. please check the logs if that's the case - if so, please update the flow of the reconciliation to prevent such conflicts.

Reply:
what do you mean by conflict? can you give more context?

Reply:
please refer to this comment: https://github.com/TykTechnologies/tyk-operator/pull/719#discussion_r1661993294